databuilder/databuilder/task/search/search_metadata_to_elasticsearch_task.py

```diff
@@ -0,0 +1,132 @@
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.transformer.base_transformer import NoopTransformer, Transformer
+from databuilder.utils.closer import Closer
+
+from databuilder.task.search.document_mappings import RESOURCE_TO_MAPPING, DefaultIndex
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SearchMetadatatoElasticasearchTask(Task):
+    
+    ENTITY_TYPE = 'doc_type'
+    ELASTICSEARCH_CLIENT_CONFIG_KEY = 'client'
+    ES_CONNECTION = 'connection'
+    CUSTOM_INDEX_CLASS = 'custom_index'
+    MAPPING_CLASS = 'document_mapping'
+    ELASTICSEARCH_NEW_INDEX_CONFIG_KEY = 'new_index'
+    ELASTICSEARCH_ALIAS_CONFIG_KEY = 'alias'
+    ELASTICSEARCH_PUBLISHER_BATCH_SIZE = 'batch_size'
+
+    def __init__(self,
+                 extractor: Extractor,
+                 transformer: Transformer = NoopTransformer()) -> None:
+        self.extractor = extractor
+        self.transformer = transformer
+
+        self._closer = Closer()
+        self._closer.register(self.extractor.close)
+        self._closer.register(self.transformer.close)
+
+    def init(self, conf: ConfigTree) -> None:
+        self.extractor.init(Scoped.get_scoped_conf(conf, self.extractor.get_scope()))
+        self.transformer.init(Scoped.get_scoped_conf(conf, self.transformer.get_scope()))
+
+        conf = Scoped.get_scoped_conf(conf, self.get_scope())
+
+        self.entity = conf.get_string(SearchMetadatatoElasticasearchTask.ENTITY_TYPE, default='table').lower()
+        self.elasticsearch_client = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_CLIENT_CONFIG_KEY, None)
+            
+        self.elasticsearch_new_index = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_NEW_INDEX_CONFIG_KEY)
+        self.elasticsearch_alias = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_ALIAS_CONFIG_KEY)
+        self.index_class = conf.get(SearchMetadatatoElasticasearchTask.CUSTOM_INDEX_CLASS, DefaultIndex)
+        self.document_mapping = conf.get(SearchMetadatatoElasticasearchTask.MAPPING_CLASS, RESOURCE_TO_MAPPING[self.entity])
+        
+        self.elasticsearch_batch_size = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_PUBLISHER_BATCH_SIZE,
+                                                      10000)
+
+    def to_document(self, document_mapping: Document, metadata: Dict, index: str) -> Document:
+        return document_mapping(_index=index, **metadata)
+
+    def generator(self, record: Iterator) -> Dict:
+        # iterate through records
+        while record:
+            record = self.transformer.transform(record)
+            if not record:
+                # Move on if the transformer filtered the record out
+                record = self.extractor.extract()
+                continue
+            yield self.to_document(document_mapping=self.document_mapping,
+                                   metadata=record,
+                                   index=self.elasticsearch_new_index).to_dict(True)
```

Comment:
do we still need to_dict?

Reply:
yes because bulk is not part of elasticsearch-dsl